src/com/amazon/kinesis/streaming/agent/tailing/SourceFile.java

```diff
@@ -114,6 +115,13 @@ public String toString() {
     private void validateDirectory(Path dir) {
         Preconditions.checkArgument(dir != null, "Directory component is empty!");
         // TODO: validate that the directory component has no glob characters
+
+        //check the permissions of every level in the directory path
+        for(Path level : dir.toAbsolutePath()){
```

Comment:
I don't think dir.toAbsolutePath() will give you a list of levels, so no point of iteration here.